index.js

```diff
@@ -282,4 +282,38 @@ Redular.prototype.getEventExpiry = async function (eventKey) {
     }
 };
 
+/**
+ * Get events that fall in the inclusive range of startDate, endDate
+ * @param {Date} startDate - Starting date of range (inclusive)
+ * @param {Date} endDate - Ending date of range (inclusive)
+ * @returns {Array<String>} - List of event keys
+ */
+Redular.prototype.getEvents = async function (startDate, endDate) {
+    try {
+        let datesInRange = [];
+
+        // Scan to get keys
+        let scanResult = await this.redis.promisfyCommand('SCAN', ['0', 'MATCH', 'redular:*']);
```

Comment:
do while doesn't make sense if you're just going to do this outside and inside the block. A regular while loop does the same thing